### PR TITLE
`azurerm_application_insights` - support for `sampling_percentage`

### LINKED ISSUE #4487: Support sampling_percentage In azurerm_application_insights
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

This is a feature request for the ability to manage the sampling rate of an application insights resource using the azurerm provider. 

It is currently possible to set the sampling percentage using the UI/ARM templates/AZ PowerShell/etc, see the references section below. It would be nice to be able to control it from TF, too.

### New or Affected Resource(s)

* azurerm_application_insights

### Potential Terraform Configuration

```hcl
resource "azurerm_application_insights" "example" {
  name                           = "example"
  location                       = "eastus2"
  resource_group_name            = azurerm_resource_group.example.name
  application_type               = "web"
  sampling_percentage            = var.env == "prod" ? 100 : 25
}
```

### References

Some other ways of creating application insights instances, these all support sampling percentage
* ARM https://docs.microsoft.com/en-us/azure/templates/microsoft.insights/2015-05-01/components
* AZ PowerShell https://docs.microsoft.com/en-us/powershell/module/az.applicationinsights/new-azapplicationinsights?view=azps-2.7.0

### PR DESCRIPTION
**Pull request by aqche**
Fixes #4487 

Adds the `sampling_percentage` argument for the `azurerm_application_insights`  resource.

```
--- PASS: TestAccAzureRMApplicationInsights_complete (92.46s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm       92.504s
```

### CODE REVIEW

File: azurerm/resource_arm_application_insights.go [hunk 1]
```diff
@@ -176,6 +185,10 @@ func resourceArmApplicationInsightsRead(d *schema.ResourceData, meta interface{}
 		d.Set("application_type", string(props.ApplicationType))
 		d.Set("app_id", props.AppID)
 		d.Set("instrumentation_key", props.InstrumentationKey)
+
+		if v := props.SamplingPercentage; v != nil {
+			d.Set("sampling_percentage", v)
+		}
```
[the remark this answers is not in the document]
Comment (aqche) [reply azurerm/resource_arm_application_insights.go]: sounds good, I applied the suggestion!

File: azurerm/resource_arm_application_insights.go [hunk 2]
```diff
@@ -111,6 +116,10 @@ func resourceArmApplicationInsightsCreateUpdate(d *schema.ResourceData, meta int
 		ApplicationType: insights.ApplicationType(applicationType),
 	}
 
+	if v, ok := d.GetOk("sampling_percentage"); ok {
```
[the remark this answers is not in the document]
Comment (aqche) [reply azurerm/resource_arm_application_insights.go]: good to know, making 100% the default sounds good to me

**Comment by aqche** [comment]
@tombuildsstuff thanks for the feedback, pushed up updates and build is passing. feel free to take another look when you have the chance!

File: azurerm/resource_arm_application_insights.go [hunk 3]
```diff
@@ -63,6 +63,12 @@ func resourceArmApplicationInsights() *schema.Resource {
 				}, true),
 			},
 
+			"sampling_percentage": {
+				Type:     schema.TypeFloat,
+				Optional: true,
+				Default:  100,
+			},
```
Comment (katbyte) [inline azurerm/resource_arm_application_insights.go]: As this is a percentage could we validate it is between 0 and 100?

**Comment by aqche** [comment]
@katbyte thanks for the review, added the validation and looks like the build is passing. Let me know what you think!

**Pull request merged**